src/components/fields/image-upload/image-manager/image-manager.ts

```diff
@@ -106,11 +132,13 @@ export const ImageManager = (props: IProps) => {
 					case EImageStatus.CONVERTED:
 					case EImageStatus.RECOMPRESSED:
 						if (!editImage) {
+							const shouldPreventDefault = !dispatchFieldEvent("upload-done", id, { imageData: image });
```

Comment:
`upload-done` sounds like the image has already been uploaded, suggest to rename as `before-upload` or `upload-ready`